src/Winium.Desktop.Driver/CommandExecutors/IsElementSelectedExecutor.cs

```diff
@@ -19,6 +19,12 @@ protected override string DoImpl()
 
             var element = this.Automator.Elements.GetRegisteredElement(registeredKey);
 
+            var controlType = element.GetAutomationPropertyValue<ControlType>(AutomationElement.ControlTypeProperty);
+            if (controlType.Equals(ControlType.CheckBox))
```

Comment:
Maybe we should not hard check if type is `CheckBox`? I think we can make it more agile by checking if element does support `TogglePattern`, and if it does, then returning value of `ToggleStateProperty`. This will automatically add support for custom elements that do implement `TogglePattern`, like buttons with state, etc. 

But we should decided on priorities, it is possible that one element can implement both `TogglePattern ` and `SelectionItemPattern `, we should decided on which one of them should take precedence.

----

А может тут не приводить явно к `CheckBox`, а сделать более гибкий метод, который будет проверять, поддерживает ли объект `TogglePattern` и, если да, то брать у него соответствующее свойство `ToggleStateProperty`?

Причем тут надо понять порядок приоритетов (что вызывать первым), вдруг один элемент может не только `TogglePattern` реализовать, но и `SelectionItemPattern`.